From 2ebc67ff8ca6a8906c52499cb67e958ba1931f0f Mon Sep 17 00:00:00 2001
From: Lawrence Rust <lvr@softsystem.co.uk>
Date: Sat, 29 Oct 2011 12:23:50 +0200
Subject: [PATCH 041/124] libmythtv: Correct test for MPEG TS continuity
 counter

The transport stream continuity counter is only incremented iff the packet
has a payload.  This change only reports a continuity error iff the
packet has a payload.

Signed-off-by: Lawrence Rust <lvr@softsystem.co.uk>
---
 mythtv/libs/libmythtv/dtvrecorder.cpp |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/mythtv/libs/libmythtv/dtvrecorder.cpp b/mythtv/libs/libmythtv/dtvrecorder.cpp
index 347df7c..8db0805 100644
--- a/mythtv/libs/libmythtv/dtvrecorder.cpp
+++ b/mythtv/libs/libmythtv/dtvrecorder.cpp
@@ -1278,7 +1278,7 @@ bool DTVRecorder::ProcessTSPacket(const TSPacket &tspacket)
 
     // Check continuity counter
     uint old_cnt = _continuity_counter[pid];
-    if ((pid != 0x1fff) && !CheckCC(pid, tspacket.ContinuityCounter()))
+    if ((pid != 0x1fff) && tspacket.HasPayload() && !CheckCC(pid, tspacket.ContinuityCounter()))
     {
         int v = _continuity_error_count.fetchAndAddRelaxed(1) + 1;
         double erate = v * 100.0 / _packet_count.fetchAndAddRelaxed(0);
@@ -1382,7 +1382,7 @@ bool DTVRecorder::ProcessAVTSPacket(const TSPacket &tspacket)
 
     // Check continuity counter
     uint old_cnt = _continuity_counter[pid];
-    if ((pid != 0x1fff) && !CheckCC(pid, tspacket.ContinuityCounter()))
+    if ((pid != 0x1fff) && tspacket.HasPayload() && !CheckCC(pid, tspacket.ContinuityCounter()))
     {
         int v = _continuity_error_count.fetchAndAddRelaxed(1) + 1;
         double erate = v * 100.0 / _packet_count.fetchAndAddRelaxed(0);
-- 
1.7.9.5

